python/ray/util/sgd/v2/session.py

```diff
@@ -226,3 +256,54 @@ def train_func():
     """
     session = get_session()
     return session.world_rank
+
+
+def load_checkpoint() -> Optional[Dict]:
+    """Loads checkpoint data onto the worker.
+
+    .. code-block:: python
+
+        from ray.util import sgd
+
+        def train_func():
+            checkpoint = sgd.load_checkpoint()
+            for iter in range(checkpoint["epoch"], 5):
+                print(iter)
+
+        trainer = Trainer(backend="torch")
+        trainer.start()
+        trainer.run(train_func, checkpoint={"epoch": 3})
+        # 3
+        # 4
+        trainer.shutdown()
+
+    Args:
+        **kwargs: Any key value pair to be checkpointed by SGD.
+    """
+    session = get_session()
+    return session.loaded_checkpoint
```

Comment:
I don't think this behavior is exactly what we want right? This will always load from the checkpoint passed in by the user. Instead this should load in from the most recently saved checkpoint. Basically `session` should keep track of the latest checkpoint in addition to sending it to the Executor.

And can you also add a test for load->save->load

Reply:
Updated with new test case!